src/ndisc/nm-lndp-ndisc.c

```diff
@@ -221,10 +221,10 @@ receive_ra (struct ndp *ndp, struct ndp_msg *msg, gpointer user_data)
 				.preferred = ndp_msg_opt_prefix_preferred_time (msg, offset),
 			};
 
-			if (address.preferred > address.lifetime)
-				address.preferred = address.lifetime;
-			if (nm_ndisc_complete_and_add_address (ndisc, &address))
-				changed |= NM_NDISC_CONFIG_ADDRESSES;
+			if (address.preferred <= address.lifetime) {
+				if (nm_ndisc_complete_and_add_address (ndisc, &address))
+					changed |= NM_NDISC_CONFIG_ADDRESSES;
+			}
```

Comment:
Should we also check 5.5.3.b:

>     b)  If the prefix is the link-local prefix, silently ignore the
>       Prefix Information option.

? Especially because later we assert:

```
	nm_assert (!IN6_IS_ADDR_UNSPECIFIED (&new->address));
	nm_assert (!IN6_IS_ADDR_LINKLOCAL (&new->address));
```

Reply:
I think that is done already:
- for RAs: https://github.com/NetworkManager/NetworkManager/blob/cec7ade86c2ac2a20a5b3cdabcf375ece347dd76/src/ndisc/nm-lndp-ndisc.c#L198
- for shared-mode: https://github.com/NetworkManager/NetworkManager/blob/cec7ade86c2ac2a20a5b3cdabcf375ece347dd76/src/devices/nm-device.c#L3548

Indeed, the latter doesn't cover `IN6_IS_ADDR_UNSPECIFIED()`. Adding a patch for that...